here-naksha-storage-http/src/main/java/com/here/naksha/storage/http/connector/ConnectorInterfaceReadExecute.java

```diff
@@ -18,54 +18,123 @@
  */
 package com.here.naksha.storage.http.connector;
 
-import static com.here.naksha.common.http.apis.ApiParamsConst.FEATURE_IDS;
+import static com.here.naksha.common.http.apis.ApiParamsConst.*;
 
 import com.here.naksha.lib.core.NakshaContext;
+import com.here.naksha.lib.core.models.geojson.WebMercatorTile;
+import com.here.naksha.lib.core.models.geojson.coordinates.BBox;
 import com.here.naksha.lib.core.models.geojson.implementation.XyzFeatureCollection;
 import com.here.naksha.lib.core.models.naksha.Space;
 import com.here.naksha.lib.core.models.payload.Event;
-import com.here.naksha.lib.core.models.payload.events.feature.GetFeaturesByIdEvent;
+import com.here.naksha.lib.core.models.payload.events.PropertyQueryOr;
+import com.here.naksha.lib.core.models.payload.events.feature.*;
+import com.here.naksha.lib.core.models.storage.POp;
+import com.here.naksha.lib.core.models.storage.ReadFeatures;
 import com.here.naksha.lib.core.models.storage.ReadFeaturesProxyWrapper;
 import com.here.naksha.lib.core.models.storage.Result;
 import com.here.naksha.lib.core.util.json.JsonSerializable;
 import com.here.naksha.storage.http.PrepareResult;
 import com.here.naksha.storage.http.RequestSender;
 import java.net.http.HttpResponse;
 import java.util.List;
+import org.apache.commons.lang3.NotImplementedException;
 import org.jetbrains.annotations.NotNull;
 
 public class ConnectorInterfaceReadExecute {
 
   @NotNull
   public static Result execute(NakshaContext context, ReadFeaturesProxyWrapper request, RequestSender sender) {
-    return switch (request.getReadRequestType()) {
-      case GET_BY_IDS -> executeFeaturesByIds(context, request, sender);
-      default -> throw new IllegalStateException("Unexpected value: " + request.getReadRequestType());
-    };
-  }
+    String streamId = context.getStreamId();
+    String dataHubSpaceName = request.getCollections().get(0);
+    Space dataHubSpace = new Space(dataHubSpaceName);
 
-  private static Result executeFeaturesByIds(
-      NakshaContext context, ReadFeaturesProxyWrapper request, RequestSender sender) {
-    Event event = createFeaturesByIdsEvent(request, context.getStreamId());
+    Event event =
+        switch (request.getReadRequestType()) {
+          case GET_BY_ID -> createFeatureByIdEvent(request);
+          case GET_BY_IDS -> createFeaturesByIdsEvent(request);
+          case GET_BY_BBOX -> createFeatureByBBoxEvent(request);
+          case GET_BY_TILE -> createFeaturesByTileEvent(request);
+          case ITERATE -> createIterateEvent(request);
+          default -> throw new IllegalStateException("Unexpected value: " + request.getReadRequestType());
```

Comment:
Since at this stage we support all the possible enum values, we can get rid of default completely.
I'll delete it.
But you are right and in the future I'll implement your approach if it will be needed.